Sources/PointFree/Pricing.swift

```diff
@@ -8,22 +8,112 @@ import HttpPipelineHtmlSupport
 import Optics
 import Prelude
 import Styleguide
+import Tuple
 
-enum PricingType {
-  case individual(BillingType)
-  case team(count: Int)
+public enum Pricing: Codable, DerivePartialIsos {
+  case individual(Billing)
+  case team(Int)
 
-  enum BillingType {
+  public static let `default` = individual(.monthly)
+
+  public static let validTeamQuantities = 2..<100
+
+  public enum Billing: String, Codable {
     case monthly
     case yearly
   }
+
+  private enum CodingKeys: String, CodingKey {
+    case individual
+    case team
+    case lane
+  }
+
+  private enum Lane: String, Codable {
+    case individual
+    case team
+  }
+
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    let billing = try container.decodeIfPresent(Billing.self, forKey: .individual)
+    let quantity = try container.decodeIfPresent(Int.self, forKey: .team)
+    let lane = try container.decodeIfPresent(Lane.self, forKey: .lane)
+
+    if let lane = lane, let billing = billing, let quantity = quantity {
+      self = lane == .individual ? .individual(billing) : .team(quantity)
+    } else if let billing = billing {
+      self = .individual(billing)
+    } else if let quantity = quantity {
+      self = .team(quantity)
+    } else {
+      throw unit // FIXME
+    }
+  }
+
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    switch self {
+    case let .individual(billing):
+      try container.encode(billing, forKey: .individual)
+    case let .team(quantity):
+      try container.encode(quantity, forKey: .team)
+    }
+  }
+
+  var plan: Stripe.Plan.Id {
+    switch self {
+    case .individual(.monthly):
+      return .init(unwrap: "individual-monthly")
+    case .individual(.yearly):
+      return .init(unwrap: "individual-yearly")
+    case .team:
+      return .init(unwrap: "team-yearly")
+    }
+  }
+
+  var quantity: Int {
+    switch self {
+    case .individual:
+      return 1
+    case let .team(quantity):
+      return quantity
+    }
+  }
+
+  var billing: Billing {
+    switch self {
+    case let .individual(billing):
+      return billing
+    case .team:
+      return .yearly
+    }
+  }
+
+  var isIndividual: Bool {
+    switch self {
+    case .individual:
+      return true
+    case .team:
+      return false
+    }
+  }
+
+  var isTeam: Bool {
+    switch self {
+    case .team:
+      return true
+    case .individual:
+      return false
+    }
+  }
 }
 
-let pricingResponse: Middleware<StatusLineOpen, ResponseEnded, Stripe.Plan.Id, Data> =
+let pricingResponse: Middleware<StatusLineOpen, ResponseEnded, Tuple2<Pricing, Database.User?>, Data> =
   writeStatus(.ok)
     >-> respond(pricingView)
```

Comment:
if you `pricingView.contramap(lower)` you can deal with regular tuples in the view and get destructuring:

```swift
private let pricingView = View<(Pricing, Database.User?)> { pricing, user in

```